include/geode/mesh/core/solid_mesh.h

```diff
@@ -293,54 +283,40 @@ namespace geode
         absl::optional< local_index_t > vertex_in_polyhedron(
             index_t polyhedron_id, index_t vertex_id ) const;
 
-        /*!
-         * Return the index in the mesh of a given polyhedron facet vertex.
-         * @param[in] polyhedron_facet_vertex Local index of the vertex in the
-         * facet of a polyhedron.
-         */
-        index_t polyhedron_facet_vertex(
-            const PolyhedronFacetVertex& polyhedron_facet_vertex ) const;
-
-        PolyhedronVertex polyhedron_facet_vertex_id(
+        PolyhedronVertex polyhedron_vertex(
             const PolyhedronFacetVertex& polyhedron_facet_vertex ) const;
 
         /*!
          * Return the indices in the mesh of the two polyhedron edge vertices.
          * @param[in] polyhedron_facet_edge Local index of edge in a polyhedron.
          */
-        std::array< index_t, 2 > polyhedron_facet_edge_vertices(
+        EdgeVertices polyhedron_facet_edge_vertices(
             const PolyhedronFacetEdge& polyhedron_facet_edge ) const;
 
-        absl::optional< PolyhedronFacetEdge >
-            polyhedron_facet_edge_from_vertices(
-                const std::array< index_t, 2 >& edge_vertices ) const;
+        absl::optional< PolyhedronFacetEdge > polyhedron_facet_edge(
+            const EdgeVertices& edge_vertices,
+            absl::optional< index_t > polyhedron_id ) const;
 
-        absl::optional< PolyhedronFacetEdge >
-            polyhedron_facet_edge_from_vertices(
-                const std::array< index_t, 2 >& edge_vertices,
-                index_t polyhedron_id ) const;
-
-        virtual PolyhedronEdgesVertices polyhedron_edges_vertices(
+        virtual EdgesVertices polyhedron_edges_vertices(
             index_t polyhedron ) const;
 
-        absl::optional< PolyhedronFacet > polyhedron_facet_from_vertices(
-            PolyhedronFacetVertices polyhedron_facet_vertices ) const;
+        absl::optional< PolyhedronFacet > polyhedron_facet(
+            FacetVertices polyhedron_facet_vertices ) const;
 
-        PolyhedronFacetVertices polyhedron_facet_vertices(
+        FacetVertices facet_vertices(
             const PolyhedronFacet& polyhedron_facet ) const;
 
-        virtual PolyhedronFacetsVertices polyhedron_facets_vertices(
-            index_t polyhedron ) const;
+        virtual FacetsVertices facets_vertices( index_t polyhedron ) const;
 
-        virtual PolyhedronFacets polyhedron_vertex_facets(
+        virtual PolyhedronFacets vertex_polyhedron_facets(

```

Comment:
I do not understand that we get

Reply:
is it all the PolyhedronFacets with this vertex? only in given polyhedron or all polyhedra or all polyhedra around vertex starting with given PolyhedronVertex?

Reply:
PolyhedronFacets incident_polyhedron_facets_in_polyhedron(
            index_t polyhedron_id, index_t vertex_id ) const;

better?

Reply:
yes